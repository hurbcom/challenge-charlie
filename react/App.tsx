import React, { Suspense } from "react";
import ErrorBoundary from "./ErrorBoundary";
import { GlobalStyle } from "./styles/global";
import HeroImage from "@components/HeroImage";
import WeatherAccordeon from "@components/WeatherAccordeon";

const App: React.FC = () => {
    return (
        <Suspense fallback={<div>Suspending...</div>}>
            <ErrorBoundary>
                <GlobalStyle />
                <HeroImage
                    image={
                        "https://www.bing.com/th?id=OHR.KitsAspen_PT-BR8299899730_1920x1080.jpg&rf=LaDigue_1920x1080.jpg&pid=hp"
                    }
                >
                    <WeatherAccordeon />
                </HeroImage>
            </ErrorBoundary>
        </Suspense>
    );
};

export default App;
