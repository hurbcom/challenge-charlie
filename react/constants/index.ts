export const POSITIONS_TO_OPACITIES: Record<number, string> = {
    0: "--max-bg-opacity",
    1: "--medium-bg-opacity",
    2: "--min-bg-opacity",
};

export const SIZE = {
    mobileS: "320px",
    mobileM: "375px",
    mobileL: "425px",
    tablet: "768px",
    laptop: "1024px",
    laptopL: "1440px",
    desktop: "2560px",
};

export const DEVICES = {
    mobileS: `(min-width: ${SIZE.mobileS})`,
    mobileM: `(min-width: ${SIZE.mobileM})`,
    mobileL: `(min-width: ${SIZE.mobileL})`,
    tablet: `(min-width: ${SIZE.tablet})`,
    laptop: `(min-width: ${SIZE.laptop})`,
    laptopL: `(min-width: ${SIZE.laptopL})`,
    desktop: `(min-width: ${SIZE.desktop})`,
    desktopL: `(min-width: ${SIZE.desktop})`,
};
