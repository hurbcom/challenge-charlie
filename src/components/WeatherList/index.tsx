import styled from "styled-components";
import WeatherListItem, {
    WeatherItemDataType,
} from "components/WeatherListItem";

const WeatherListBase = styled.div`
    display: flex;
    flex-grow: 1;
    flex-direction: column;

    box-shadow: 0 1px 1px hsl(0deg 0% 0% / 0.05),
        0 2px 2px hsl(0deg 0% 0% / 0.05), 0 4px 4px hsl(0deg 0% 0% / 0.05),
        0 8px 8px hsl(0deg 0% 0% / 0.05), 0 16px 16px hsl(0deg 0% 0% / 0.05);

    border-bottom-left-radius: 8px;
    border-bottom-right-radius: 8px;
    overflow: hidden;
`;

type WeatherListPropsTytpe = {
    data: WeatherItemDataType[];
};

const WeatherList = ({ data }: WeatherListPropsTytpe) => {
    return (
        <WeatherListBase>
            {data.map((dailyForeacst: WeatherItemDataType, index) => (
                <WeatherListItem
                    key={`w-l--i${index + 1}`}
                    compact={index !== 0}
                    data={dailyForeacst}
                />
            ))}
        </WeatherListBase>
    );
};

export default WeatherList;
