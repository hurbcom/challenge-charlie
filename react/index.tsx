import React, { StrictMode } from "react";
import { hydrateRoot } from "react-dom/client";
import App from "./App";

const root = document.getElementById("react-app");

if (root) {
    hydrateRoot(
        root,
        <StrictMode>
            <App />
        </StrictMode>
    );
}
