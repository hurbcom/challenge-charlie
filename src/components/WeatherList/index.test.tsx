import { render, screen } from "@testing-library/react";
import WeatherList from ".";

describe("<WeatherList />", () => {
    test("render the weather list with 1 item", () => {
        const { asFragment } = render(
            <WeatherList
                data={[
                    {
                        day: "Hoje ",
                        humidity: 70,
                        pressure: 1015,
                        temperature: 24.28,
                        windDirection: "SW",
                        windSpeed: 3.39,
                        weatherDescription: "chuva leve",
                    },
                ]}
            />
        );

        expect(asFragment).toMatchSnapshot();
        expect(screen.queryAllByRole("presentation").length).toEqual(1);
    });
    test("render the weather list with 5 item", () => {
        const { asFragment } = render(
            <WeatherList
                data={[
                    {
                        day: "Hoje ",
                        humidity: 70,
                        pressure: 1015,
                        temperature: 24.28,
                        windDirection: "SW",
                        windSpeed: 3.39,
                        weatherDescription: "chuva leve",
                    },
                    {
                        day: "Amanhã ",
                        humidity: 70,
                        pressure: 1015,
                        temperature: 24.28,
                        windDirection: "SW",
                        windSpeed: 3.39,
                        weatherDescription: "ensolarado",
                    },
                    {
                        day: "15 de abril ",
                        humidity: 70,
                        pressure: 1015,
                        temperature: 24.28,
                        windDirection: "SW",
                        windSpeed: 3.39,
                        weatherDescription: "poucas nuvens",
                    },
                    {
                        day: "16 de abril ",
                        humidity: 70,
                        pressure: 1015,
                        temperature: 24.28,
                        windDirection: "SW",
                        windSpeed: 3.39,
                        weatherDescription: "chuva leve",
                    },
                    {
                        day: "17 de abril ",
                        humidity: 70,
                        pressure: 1015,
                        temperature: 24.28,
                        windDirection: "SW",
                        windSpeed: 3.39,
                        weatherDescription: "chuva leve",
                    },
                ]}
            />
        );

        expect(asFragment).toMatchSnapshot();
        expect(screen.queryAllByRole("presentation").length).toEqual(5);
        expect(screen.queryByText(/ensolarado/)).not.toBeInTheDocument();
        expect(screen.queryByText(/Amanhã/)).toBeInTheDocument();
        expect(screen.queryByText(/poucas nuvens/)).not.toBeInTheDocument();
    });
});
