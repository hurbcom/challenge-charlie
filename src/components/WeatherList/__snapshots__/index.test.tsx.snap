// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherList /> render the weather list with 1 item 1`] = `[Function]`;

exports[`<WeatherList /> render the weather list with 5 item 1`] = `[Function]`;
